lib/src/test/java/com/auth0/jwt/JWTCreatorTest.java

```diff
@@ -53,6 +53,49 @@ public void shouldAddHeaderClaim() throws Exception {
         assertThat(headerJson, JsonMatcher.hasEntry("asd", 123));
     }
 
+    @Test
+    public void shouldReturnBuilderIfNullMapIsProvided() throws Exception {
+        String signed = JWTCreator.init()
+                                  .withHeader(null)
+                                  .sign(Algorithm.HMAC256("secret"));
+
+        assertThat(signed, is(notNullValue()));
+    }
+
+    @Test
+    public void shouldOverwriteExistingIfHeadersMapContainsTheSameKey() throws Exception {
```

Comment:
This test looks confusing as the name tries to tell that a header claim with the same name was already set and then overwritten by the `withHeader` call. However that is not true, as the `withClaim` is adding a claim to the _payload_ object (not the header). There's no overwrite here. 

You might want to review the PR description where you showed the failing scenario with the KID claim and test that instead.


```java
Map<String, Object> header = new HashMap<String, Object>();
header.put("kid", "123456");


String signed = JWTCreator.init()
                                  .withKeyId("abcdef")
                                  .withHeader(header)
                                  .sign(Algorithm.HMAC256("secret"));

//Check header.kid = 123456
```

You can even duplicate this test and check that when called in the alternate order, header.kid is "abcdef". e.g. name the test something like "should respect order of builder calls when setting the header"